website/blog/2024-07-08-building-your-semantic-layer-in-pieces.md

```diff
@@ -0,0 +1,96 @@
+---
+title: "How to build a Semantic Layer in pieces: step-by-step for busy analytics engineers"
+description: A deep-dive into the steps you can take to start building your dbt Semantic Layer _today_, without doing a big migration.
+slug: semantic-layer-in-pieces
+authors: [gwen_windflower]
+tags: [analytics craft]
+hide_table_of_contents: false
+date: 2024-07-09
+is_featured: true
+---
+
+The dbt Semantic Layer is founded on the idea that data transformation should be both _flexible_ — allowing for on-the-fly aggregations grouped and filtered by definable dimensions — as well as _version-controlled and tested_ — like any other codebase you should have confidence that your transformations express your organization’s business logic correctly. Historically, you had to choose between these options, but the dbt Semantic Layer brings them together. This has required new paradigms for _how_ you express your transformations though.
+
+<!-- truncate -->
+
+Because of this, we’ve noticed when talking to dbt users that they _want_ to adopt the Semantic Layer, but feel daunted by the idea of migrating their transformations to this new paradigm. The good news is that you do _not_ need to make a huge one-time migration.
+
+We’re here to discuss another way, building a Semantic Layer in pieces, and our goal is to make sure you derive increased leverage and velocity from each step on your journey. If you’re eager to start building, but like most busy analytics engineers have limited bandwidth, this one is especially for you.
+
+## System of a Noun: deciding what happens where
+
+When you’re using the dbt Semantic Layer, you want to _minimize_ _the modeling that exists outside of dbt_. Eliminate it completely if you can. Why?
+
+- It’s **duplicative, patchy, and confusing** as discussed above.
+- It’s **less powerful**.
+- You **can’t** **test** it.
+- Depending on the tool, oftentimes you **can’t** **version control** it.
+
+What you want is a unified development flow that handles **normalized transformation in dbt models** and **dynamic denormalization in the dbt Semantic Layer**.
```

Comment:
just sugg this as i struggled to understand what dynamic denormalization meant - maybe just me though!
```suggestion
What you want is a unified development flow that handles **normalized transformation in dbt models** and **dynamic denormalization in the dbt Semantic Layer** (meaning it flexibly combines and reshapes normalized data models into different useful formats whenever you need them).
```